File: leetcode/solutions/0219.ContainsDuplicateII/containsNearbyDuplicate_test.go
package solutions

import (
	"testing"
)

func Test_0219_ContainsDuplicateII(t *testing.T) {
	result := containsNearbyDuplicate([]int{99, 99}, 2)
	expected := true

	if result != expected {
		t.Errorf("TEST FAILED! got: %#v, want: %#v.", result, expected)
	}
}

File: leetcode/solutions/0219.ContainsDuplicateII/containsNearbyDuplicate_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import containsNearbyDuplicate from "./containsNearbyDuplicate.ts";

Deno.test("0219. Contains Duplicate II", () => {
  assertEquals(containsNearbyDuplicate([1, 2, 3, 1], 3), true);
  assertEquals(containsNearbyDuplicate([1, 0, 1, 1], 1), true);
  assertEquals(containsNearbyDuplicate([1, 2, 3, 1, 2, 3], 2), false);
  assertEquals(containsNearbyDuplicate([1, 2, 1], 0), false);
  assertEquals(containsNearbyDuplicate([4, 1, 2, 3, 1, 5], 3), true);
});
